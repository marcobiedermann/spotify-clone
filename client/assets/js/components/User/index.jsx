import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';

class User extends PureComponent {
  static propTypes = {
    display_name: PropTypes.string,
    followers: PropTypes.shape({
      total: PropTypes.number,
    }),
    images: PropTypes.arrayOf(
      PropTypes.shape({
        url: PropTypes.string,
      }),
    ),
  };

  static defaultProps = {
    display_name: '',
    followers: {
      total: 0,
    },
    images: [],
  };

  render() {
    const { display_name, followers, images } = this.props;

    return (
      <div>
        <h1>{display_name}</h1>
        {images[0] && (
          <figure>
            <img src={images[0].url} alt={display_name} />
          </figure>
        )}
        {followers && <p>{followers.total} Followers</p>}
      </div>
    );
  }
}

export default User;
