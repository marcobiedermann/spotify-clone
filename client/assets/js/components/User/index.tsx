/* eslint-disable camelcase */

import React, { FC } from 'react';
import Image, { ImageProps } from '../Image';

interface Followers {
  total: number;
}

export interface UserProps {
  display_name: string;
  followers?: Followers;
  images?: ImageProps[];
}

const User: FC<UserProps> = (props) => {
  const { display_name, followers, images } = props;

  return (
    <div>
      <h1>{display_name}</h1>
      {images[0] && (
        <figure>
          <Image url={images[0].url} alt={display_name} />
        </figure>
      )}
      {followers && <p>{followers.total} Followers</p>}
    </div>
  );
};

export default User;
