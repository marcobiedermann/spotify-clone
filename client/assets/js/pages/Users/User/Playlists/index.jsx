import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import Playlists from '../../../../components/Playlists';
import PlaylistPageContainer from '../../../../containers/UserPlaylistPage';

class PlaylistsPage extends Component {
  componentDidMount() {
    const { accessToken, fetchUserPlaylists } = this.props;

    fetchUserPlaylists(accessToken, '11168662039');
  }

  render() {
    const { match, playlists } = this.props;

    return (
      <Switch>
        <Route path={`${match.url}/:playlist_id`} component={PlaylistPageContainer} />
        <Route
          path={`${match.url}`}
          component={() => (
            <Playlists playlists={playlists} />
          )}
        />
      </Switch>
    );
  }
}

PlaylistsPage.propTypes = {
  accessToken: PropTypes.string,
  fetchUserPlaylists: PropTypes.func,
  match: PropTypes.shape({
    url: PropTypes.string,
  }),
  playlists: PropTypes.arrayOf(PropTypes.shape),
};

PlaylistsPage.defaultProps = {
  accessToken: '',
  fetchUserPlaylists: () => {},
  match: {
    url: '',
  },
  playlists: [],
};

export default PlaylistsPage;
