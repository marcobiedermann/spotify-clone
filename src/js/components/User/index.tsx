import React from 'react';
import Image from '../Image';

interface Followers {
  total: number;
}

interface Image {
  height: number;
  url: string;
  width: number;
}

export interface UserProps {
  display_name: string;
  followers?: Followers;
  images?: Image[];
}

function User(props: UserProps): JSX.Element {
  const { display_name, followers, images } = props;

  return (
    <div>
      <h1>{display_name}</h1>
      {images[0] && (
        <figure>
          <Image url={images[0].url} alt={display_name} />
        </figure>
      )}
      {followers && <p>{followers.total} Followers</p>}
    </div>
  );
}

export default User;
