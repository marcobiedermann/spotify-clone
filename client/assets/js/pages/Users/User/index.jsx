import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import PlaylistsPage from './Playlists';
import User from '../../../components/User';
import { ACCESS_TOKEN } from '../../../constants/config';

class UserPage extends Component {
  componentDidMount() {
    const { fetchUser, match } = this.props;

    fetchUser(ACCESS_TOKEN, match.params.user_id);
  }

  render() {
    const { match, me } = this.props;

    return (
      <Switch>
        <Route path={`${match.url}/playlists`} component={PlaylistsPage} />
        <Route
          path={`${match.url}`}
          component={() => (
            <div>
              <User {...me.me} />
            </div>
          )}
        />
      </Switch>
    );
  }
}

UserPage.propTypes = {
  fetchUser: PropTypes.func,
  match: PropTypes.shape({
    params: PropTypes.shape({
      user_id: PropTypes.string,
    }),
    url: PropTypes.string,
  }),
  me: PropTypes.shape({
    me: PropTypes.shape(),
  }),
};

UserPage.defaultProps = {
  fetchUser: () => {},
  match: {
    params: {
      user_id: '',
    },
    url: '',
  },
  me: {
    me: null,
  },
};

export default UserPage;
