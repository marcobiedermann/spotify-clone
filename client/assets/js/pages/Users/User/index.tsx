import React, { Component } from 'react';
import { Helmet } from 'react-helmet';
import { connect } from 'react-redux';
import { RouteChildrenProps, useParams } from 'react-router-dom';
import { bindActionCreators } from 'redux';
import { fetchUser } from '../../../actions/users';
import Error from '../../../components/Error';
import Loader from '../../../components/Loader';
import User from '../../../components/User';

export interface UserPageProps extends RouteChildrenProps {
  accessToken: string;
  error: {
    message: string;
  };
  fetchUser: any;
  isLoading: boolean;
  me: {
    me: {
      display_name: string;
    };
  };
}

export class UserPage extends Component<UserPageProps> {
  componentDidMount() {
    const { accessToken, fetchUser } = this.props;
    const { userId } = useParams();

    fetchUser(accessToken, userId);
  }

  render() {
    const { error, isLoading, me } = this.props;

    if (error) {
      return <Error>{error.message}</Error>;
    }

    if (isLoading) {
      return <Loader />;
    }

    return (
      <>
        <Helmet>
          <title>{me.me.display_name}</title>
        </Helmet>
        <User {...me.me} />
      </>
    );
  }
}

const mapStateToProps = (state) => ({
  ...state,
  error: state.users.error,
  isLoading: state.users.isLoading,
  user: state.users.user,
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      fetchUser,
    },
    dispatch,
  );

export default connect(mapStateToProps, mapDispatchToProps)(UserPage);
