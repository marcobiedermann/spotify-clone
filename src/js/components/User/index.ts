import User from './User';

export default User;
export * from './User';
