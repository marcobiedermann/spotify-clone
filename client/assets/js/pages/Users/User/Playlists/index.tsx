import React, { Component } from 'react';
import { Helmet } from 'react-helmet';
import { connect } from 'react-redux';
import { RouteChildrenProps } from 'react-router-dom';
import { bindActionCreators } from 'redux';
import { fetchUserPlaylists } from '../../../../actions/users';
import Error from '../../../../components/Error';
import Loader from '../../../../components/Loader';
import Playlists, { PlaylistsProps } from '../../../../components/Playlists';

export interface PlaylistsPageProps extends RouteChildrenProps {
  accessToken: string;
  error: {
    message: string;
  };
  fetchUserPlaylists: any;
  isLoading: boolean;
  playlists: PlaylistsProps;
}

export class PlaylistsPage extends Component<PlaylistsPageProps> {
  componentDidMount() {
    const { accessToken, fetchUserPlaylists } = this.props;

    fetchUserPlaylists(accessToken, '11168662039');
  }

  render() {
    const { error, isLoading, playlists } = this.props;

    if (error) {
      return <Error>{error.message}</Error>;
    }

    if (isLoading) {
      return <Loader />;
    }

    return (
      <>
        <Helmet>
          <title>Playlists</title>
        </Helmet>
        <Playlists {...playlists} />
      </>
    );
  }
}

const mapStateToProps = (state) => ({
  ...state,
  error: state.users.error,
  isLoading: state.users.isLoading,
  playlists: state.users.playlists,
});

const mapDispatchToProps = (dispatch) =>
  bindActionCreators(
    {
      fetchUserPlaylists,
    },
    dispatch,
  );

export default connect(mapStateToProps, mapDispatchToProps)(PlaylistsPage);
