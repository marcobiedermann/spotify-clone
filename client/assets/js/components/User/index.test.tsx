import React from 'react';
import renderer from 'react-test-renderer';
import User from '.';
import userFixtures from './__fixtures__';

it('renders correctly', () => {
  const tree = renderer.create(<User {...userFixtures} />).toJSON();

  expect(tree).toMatchSnapshot();
});
