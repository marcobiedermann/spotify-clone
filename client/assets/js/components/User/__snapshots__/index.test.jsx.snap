// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h1>
    
  </h1>
</div>
`;
